io/plugins/eu.esdihumboldt.hale.io.json/src/eu/esdihumboldt/hale/io/json/writer/NamespaceManager.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022 wetransform GmbH
+ * 
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     wetransform GmbH <http://www.wetransform.to>
+ */
+
+package eu.esdihumboldt.hale.io.json.writer;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+
+/**
+ * Abstract class to manage Namespaces. This class can be further extended to
+ * manage and include namespace prefixes in the exported Json or geoJson
+ * formats. Not Thread safe.
+ * 
+ * @author Kapil Agnihotri
+ */
+public abstract class NamespaceManager {
```

Comment:
I think a it better would have been to introduce an interface that only holds the methods that `InstanceToJson` actually uses. That would have made the `IgnoreNamespaces` implementation simpler and it would not need to carry the unused map around.

But the solution as you implemented is also totally fine :+1:

Reply:
First, I made it an interface but then thought of making it abstract 
1. to move the common code like (`getNamespaces()` and `getNamespace()`) to one place. 
2. I had an understanding that `IgnoreNamespaces` would anyways will need to return a map for `getnamespace*()`. Otherwise returning null would then lead to unchecked NPEs and a bit changes in the code.

Reply:
Makes sense. Then I did not look at the details closely enough. Thanks!

Reply:
Thinking again about it, the sole methods used seem to be `getPrefix` and `getNamespaces`. `getNamespaces` is used to add the namespace definitions to the file in case of GeoJson.
But if we ignore the namespaces, it does not make sense to define any. So `IgnoreNamespaces` could just return an empty map or `null` for `getNamespaces`, and no namespace definitions should be written to the output in that case.

Reply:
True, We don't need to return any name space definition if we are ignoring namespace. I created an interface and now returning either empty string for prefix or `null` (I guess returning an empty map is unnecessary just to avoid a null check it will also consume some memory). Also, now I added a null check when writing the namespaces incase of GeoJson.

PR ready for review.